libff/algebra/fields/fp.tcc

```diff
@@ -422,89 +464,80 @@ Fp_model<n,modulus>& Fp_model<n,modulus>::operator+=(const Fp_model<n,modulus>&
 {
 #ifdef PROFILE_OP_COUNTS
     this->sub_cnt++;
 #endif
 #if defined(__x86_64__) && defined(USE_ASM)
-    if (n == 3)
-    {
-        __asm__
-            (SUB_FIRSTSUB
-             SUB_NEXTSUB(8)
-             SUB_NEXTSUB(16)
-
-             "jnc     done%=\n\t"
-
-             SUB_FIRSTADD
-             SUB_NEXTADD(8)
-             SUB_NEXTADD(16)
-
-             "done%=:\n\t"
-             :
-             : [A] "r" (this->mont_repr.data), [B] "r" (other.mont_repr.data), [mod] "r" (modulus.data)
-             : "cc", "memory", "%rax");
-    }
-    else if (n == 4)
-    {
-        __asm__
-            (SUB_FIRSTSUB
-             SUB_NEXTSUB(8)
-             SUB_NEXTSUB(16)
-             SUB_NEXTSUB(24)
-
-             "jnc     done%=\n\t"
-
-             SUB_FIRSTADD
-             SUB_NEXTADD(8)
-             SUB_NEXTADD(16)
-             SUB_NEXTADD(24)
-
-             "done%=:\n\t"
-             :
-             : [A] "r" (this->mont_repr.data), [B] "r" (other.mont_repr.data), [mod] "r" (modulus.data)
-             : "cc", "memory", "%rax");
-    }
-    else if (n == 5)
-    {
-        __asm__
-            (SUB_FIRSTSUB
-             SUB_NEXTSUB(8)
-             SUB_NEXTSUB(16)
-             SUB_NEXTSUB(24)
-             SUB_NEXTSUB(32)
-
-             "jnc     done%=\n\t"
-
-             SUB_FIRSTADD
-             SUB_NEXTADD(8)
-             SUB_NEXTADD(16)
-             SUB_NEXTADD(24)
-             SUB_NEXTADD(32)
-
-             "done%=:\n\t"
-             :
-             : [A] "r" (this->mont_repr.data), [B] "r" (other.mont_repr.data), [mod] "r" (modulus.data)
-             : "cc", "memory", "%rax");
-    }
-    else
+    if (n == 3) {
+        __asm__(                 // Preserve alignment
+            SUB_FIRSTSUB()       //
+            SUB_NEXTSUB(8)       //
+            SUB_NEXTSUB(16)      //
+            "jnc     done%=\n\t" //
+            SUB_FIRSTADD()       //
+            SUB_NEXTADD(8)       //
+            SUB_NEXTADD(16)      //
+            "done%=:\n\t"        //
+            :
+            : [A] "r"(this->mont_repr.data),
+              [B] "r"(other.mont_repr.data),
+              [mod] "r"(modulus.data)
+            : "cc", "memory", "%rax");
+    } else if (n == 4) {
+        __asm__(                 // Preserve alignment
+            SUB_FIRSTSUB()       //
+            SUB_NEXTSUB(8)       //
+            SUB_NEXTSUB(16)      //
+            SUB_NEXTSUB(24)      //
+            "jnc     done%=\n\t" //
+            SUB_FIRSTADD()       //
+            SUB_NEXTADD(8)       //
+            SUB_NEXTADD(16)      //
+            SUB_NEXTADD(24)      //
+            "done%=:\n\t"        //
+            :
+            : [A] "r"(this->mont_repr.data),
+              [B] "r"(other.mont_repr.data),
+              [mod] "r"(modulus.data)
+            : "cc", "memory", "%rax");
+    } else if (n == 5) {
+        __asm__(                 // Preserve alignment
+            SUB_FIRSTSUB()       //
```

Comment:
What was the issue with the missing parenthesis?

Reply:
Without the parentheses, I think clang-format would rearrange everything to make it unreadable.  IIRC, adding parentheses also made the macro consistent with several other macros.

Reply:
I see, thanks